core/drag/drag-manager.js

```diff
@@ -0,0 +1,1314 @@
+     */
+    _findDraggableAtPosition: {
+        value: function (positionX, positionY) {
+            return this._findRegisteredComponentAtPosistion(
+                positionX,
+                positionY, 
+                DRAGGABLE
+            );
+        }
+    },
+
+    /**
+     * @private
+     * @function
+     * @param {string} positionX - x coordinate
+     * @param {string} positionY - y coordinate
+     * @description try to find a droppable component at the given position.
+     * @returns {Component|null}
+     */
+    _findDroppableAtPosition: {
+        value: function (positionX, positionY) {
+            return this._findRegisteredComponentAtPosistion(
+                positionX,
+                positionY, 
+                DROPABBLE
+            );
+        }
+    },
+
+    /**
+     * @private
+     * @function
+     * @param {string} positionX - x coordinate
+     * @param {string} positionY - y coordinate
+     * @description try to find a droppable or draggable component
+     * at the given position.
+     * @returns {Component|null}
+     */
+    _findRegisteredComponentAtPosistion: {
+        value: function (positionX, positionY, role) {
+            var targetComponent = this._findComponentAtPosition(
+                positionX, positionY
+            ),
+            registeredComponent = null;
+
+            if (targetComponent) {
+                var components = role === DRAGGABLE ? 
+                    this._draggables : this._droppables,
+                    index;
+
+                while (targetComponent) {
+                    if ((index = components.indexOf(targetComponent)) > -1) {
+                        registeredComponent = components[index];
+                        targetComponent = null;
+                    } else {
+                        targetComponent = targetComponent.parentComponent;
+                    }
+                }
+            }
+
+            return registeredComponent;
+        }
+    },
+
+    /**
+     * @private
+     * @function
+     * @param {string} positionX - x coordinate
+     * @param {string} positionY - y coordinate
+     * @description try to find a component at the given position.
+     * @returns {Component|null}
+     */
+    _findComponentAtPosition: {
+        value: function (positionX, positionY) {
+            var element = document.elementFromPoint(positionX, positionY),
+                component = null;
+
+            // that is done at several place in the framework
+            // we should re-organize that 
+            while (element && !(component = element.component)) {
+                element = element.parentElement;
+            }
+
+            return component;
+        }
+    },
+
+    /**
+     * @private
+     * @function
+     * @description add translate listeners.
+     */
+    _addTranslateListeners: {
+        value: function () {
+            this._translateComposer.addEventListener('translate', this);
```

Comment:
shouldn't there be a third argument for capture/bubble? Remember in the past Firefox barfing on that